frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -0,0 +1,287 @@
+};
+
+export const createBuildConfig = (
+  formData: DeployImageFormData,
+  dryRun: boolean,
+): Promise<K8sResourceKind> => {
+  const {
+    project: { name: namespace },
+    application: { name: application },
+    name,
+    isi: { name: isi_name, tag },
+    build: { env, triggers },
+    labels: userLabels,
+  } = formData;
+
+  const defaultLabels = getAppLabels(name, application, isi_name);
+  const buildConfig = {
+    apiVersion: 'build.openshift.io/v1',
+    kind: 'BuildConfig',
+    metadata: {
+      name,
+      namespace,
+      labels: { ...defaultLabels, ...userLabels },
+    },
+    spec: {
+      output: {
+        to: {
+          kind: 'ImageStreamTag',
+          name: `${name}:latest`,
+        },
+      },
+      strategy: {
+        type: 'Source',
+        sourceStrategy: {
+          env,
+          from: {
+            kind: 'ImageStreamTag',
+            name: `${isi_name}:${tag}`,
+            namespace,
+          },
+        },
+      },
+      triggers: [
+        ...(triggers.image ? [{ type: 'ImageChange', imageChange: {} }] : []),
+        ...(triggers.config ? [{ type: 'ConfigChange' }] : []),
+      ],
+    },
+  };
+
+  return k8sCreate(BuildConfigModel, buildConfig, dryRun ? dryRunOpt : {});
+};
+
+export const createDeploymentConfig = (
+  formData: DeployImageFormData,
+  dryRun: boolean,
+): Promise<K8sResourceKind> => {
+  const {
+    project: { name: namespace },
+    application: { name: application },
+    name,
+    searchTerm,
+    isi: { name: isi_name, tag, ports },
+    deployment: { env, replicas, triggers },
+    labels: userLabels,
+  } = formData;
+
+  const defaultLabels = getAppLabels(name, application, isi_name);
+  const labels = _.isEmpty(userLabels) ? {app: application} : SelectorInput.objectify(userLabels);
+
+  const deploymentConfig = {
+    kind: 'DeploymentConfig',
+    apiVersion: 'apps.openshift.io/v1',
+    metadata: {
+      name,
+      namespace,
+      labels: { ...defaultLabels, ...userLabels },
+      annotations,
+    },
+    spec: {
+      replicas,
+      selector: labels,
+      template: {
+        metadata: {
+          labels,
+          annotations,
+        },
+        spec: {
+          volumes,
+          containers: [
+            {
+              name,
+              image: `${searchTerm}:latest`,
+              ports,
+              volumeMounts,
+              env,
+            },
+          ],
+        },
+      },
+      triggers: [
+        {
+          type: 'ImageChange',
+          imageChangeParams: {
+            automatic: triggers.image,
+            containerNames: [name],
+            from: {
+              kind: 'ImageStreamTag',
+              name: `${name}:${tag}`,
+            },
+          },
+        },
+        ...(triggers.config ? [{ type: 'ConfigChange' }] : []),
+      ],
+    },
+  };
+
+  return k8sCreate(DeploymentConfigModel, deploymentConfig, dryRun ? dryRunOpt : {});
+};
+
+export const createService = (
```

Comment:
Same as git import? Share code?

Reply:
They need to be different due to how the images are accessed.

Reply:
That might be possible but at this point I would rather leave it for a later refactoring.